.github/actions/composite/updateProtectedBranch/action.yml

```diff
@@ -0,0 +1,140 @@
+name: Update Protected Branch
+description: Create, approve, and merge a pull request against a protected branch
+
+inputs:
+  TARGET_BRANCH:
+    description: The target branch to update. This becomes the base branch of the pull request.
+    required: true
+  SOURCE_BRANCH:
+    description: If updating main, you must also provide a head branch to update main with.
+    required: false
+    default: ''
+  OS_BOTIFY_TOKEN:
+    description: GitHub token for OSBotify
+    required: true
+  GPG_PASSPHRASE:
+    description: Passphrase used to decrypt GPG key for OSBotify
+    required: true
+  SLACK_WEBHOOK:
+    description: URL of the slack webhook
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Validate target branch
+      if: ${{ !contains(fromJSON('["main", "staging", "production"]'), github.event.inputs.TARGET_BRANCH) }}
+      shell: bash
+      run: |
+        echo "Target branch must be one of ['main', 'staging', 'production]"
+        exit 1
+
+    # If updating main, SOURCE_BRANCH must not be empty
+    - name: Validate source branch
+      if: github.event.inputs.TARGET_BRANCH == 'main' && github.event.inputs.SOURCE_BRANCH == ''
+      shell: bash
+      run: |
+        echo "Cannot update main branch without specifying a source branch"
+        exit 1
+
+    # If updating staging, the source branch will always be main
+    # If updating production, the source branch will always be staging
+    - name: Set source branch
+      shell: bash
+      run: |
+        if [[ ${{ github.event.inputs.TARGET_BRANCH }} == 'staging' ]]; then
+          echo "SOURCE_BRANCH=main" >> "$GITHUB_ENV"
+        elif [[ ${{ github.event.inputs.TARGET_BRANCH }} == 'production' ]]; then
+          echo "SOURCE_BRANCH=staging" >> "$GITHUB_ENV"
+        else
+          echo "SOURCE_BRANCH=${{ github.event.inputs.SOURCE_BRANCH }}" >> "$GITHUB_ENV"
+        fi
+
+    - uses: Expensify/App/.github/actions/composite/setupGitForOSBotify@main
```

Comment:
Note: I did [an experiment](https://github.com/Andrew-Test-Org/Public-Test-Repo/actions/runs/2592981541) and confirmed that you can use nested composite actions